### DISCUSSION TITLE
Time-to-first-spike Coding in SpikeNeural Networks

### DISCUSSION THREAD

**Discussion by machuanfeng** [BindsNET/bindsnet#624] [Q&A]
Hello author, your previous response has solved a significant problem for me, but I am a novice in this field and have two other questions that I would like to ask you for advice.
1. Have you implemented TTFS encoding? I tried to write a TTFS encoding myself, but the effect was poor after replacement. It may be due to the low firing rate of neurons caused by too few pulses. Do you have any examples on your end that I can refer to
2.eth_ What is the highest accuracy rate of running the file mnist.py? I have repeatedly adjusted the parameters, and the current highest accuracy is 90%, but this is lower than the accuracy in the paper. If you could share the various parameter settings in the code, I would greatly appreciate it

**Comment by Hananel-Hazan** [comment]
1. Yes, we implement [Rank order encoding](https://github.com/BindsNET/bindsnet/blob/4d2868f6392448d188230f843eed57541eb14ab5/bindsnet/encoding/encodings.py#L159), you can use it to encode your input.
Base on your previous question #625  its seems that you need to increase your exposure intensity of your input
2.  The accuracy that was reported in the paper was the maximum accuracy we achieve, but due to last improvements in the library I believe that we need to run parameter search to find the best accuracy again.

**Reply by machuanfeng** [reply]
Thank you very much and express my sincere respect to you.
